log/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+pub extern crate log;
+
+#[macro_export]
+macro_rules! pa_log {
+    ($($t:tt)*) => (
+        $crate::log::log!($($t)*)
+    )
+}
+
+#[macro_export]
+#[cfg(feature="log_debug")]
+macro_rules! pa_debug {
+    ($($t:tt)*) => (
+        $crate::log::debug!($($t)*)
+    )
+}
+#[macro_export]
+#[cfg(not(feature="log_debug"))]
+macro_rules! pa_debug {
+    ($($t:tt)*) => ()
+}
+
+#[macro_export]
+#[cfg(feature="log_trace")]
+macro_rules! pa_trace {
+    ($($t:tt)*) => (
+        $crate::log::trace!($($t)*)
+    )
+}
+#[macro_export]
+#[cfg(not(feature="log_trace"))]
+macro_rules! pa_trace {
+    ($($t:tt)*) => ()
+}
+
+#[macro_export]
+#[cfg(feature="log_info")]
+macro_rules! pa_info {
+    ($($t:tt)*) => (
+        $crate::log::info!($($t)*)
+    )
+}
+#[macro_export]
+#[cfg(not(feature="log_info"))]
+macro_rules! pa_info {
+    ($($t:tt)*) => ()
+}
+
+#[macro_export]
+#[cfg(feature="log_warn")]
+macro_rules! pa_warn {
+    ($($t:tt)*) => (
+        $crate::log::warn!($($t)*)
+    )
+}
+#[macro_export]
+#[cfg(not(feature="log_warn"))]
+macro_rules! pa_warn {
+    ($($t:tt)*) => ()
+}
+
+#[macro_export]
+macro_rules! pa_error {
+    ($($t:tt)*) => (
+        $crate::log::error!($($t)*)
+    )
+}
```

Comment:
I am not sure about this. But I also could not work without it.

Reply:
Just because of compilation time?

Reply:
It can make things more difficult if you forget to enable the appropriate feature.
(Then again, most users would probably not have to debug pathfinder anyway.)
I ran into this myself…

Reply:
I just don't have any better solution…

Reply:
I'm not sure what you mean by forgetting to enable the appropriate feature, could you elaborate?